### PR TITLE
IaaS DNS Forwarders

### PR DESCRIPTION
**Pull request by FabienGilbert**
# Description

Template and Readme to deploy IaaS DNS Forwarders.

## Issue reference

The issue this PR will close: #745 

## Checklist

Please make sure you've completed the relevant tasks for this PR out of the following list:

* [X] All acceptance criteria in the backlog item are met
* [X] The documentation is updated to cover any new or changed features
* [X] Manual tests have passed
* [X] Relevant issues are linked to this PR

### CODE REVIEW

File: src/bicep/examples/iaas-dns-forwarders/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Azure IaaS DNS Forwarders example
```
Comment (lisamurphy-msft) [inline src/bicep/examples/iaas-dns-forwarders/README.md:1]: Please append an entry to this in the root examples folder README.md table located: 
https://github.com/Azure/missionlz/tree/main/src/bicep/examples#example-explanations

**Review by lisamurphy-msft — CHANGES_REQUESTED** [review]
Thank you for providing an example of how to deploy MLZ with an existing DNS infrastructure! 
For customers that use their own DNS, this will help provide a new interoperability! 

Two items I would like to see addressed, if possible: 

*  remove the zip file in your commit, I don't think this is needed
*  add an entry to the table in the examples README.md to provide an overview of this new example, comment already provided on this.

File: src/bicep/examples/iaas-dns-forwarders/README.md (hunk 1 above)
Comment (FabienGilbert) [reply src/bicep/examples/iaas-dns-forwarders/README.md:1]: I've added an entry for this example to the root examples folder README.md table.

**Comment by lisamurphy-msft** [comment]
Thank you for pushing the proposed change! 
This looks good and I was able to successfully test this!

**Review by lisamurphy-msft — APPROVED** [review]

**Pull request merged**